I'm somehow stuck here and there are two things left, where I need help.
1. I think I managed to separate the `CustomPagination` into its own file but now get a TS warning:
  ```
/Users/jgrumboeck/git/jgrumboe/cloud-carbon-footprint/packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx
  16:10  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

✖ 1 problem (0 errors, 1 warning)
  ```
  Don't know how to get rid of it.

2. I didn't manage to refactor the function parameters into props. Tried but didn't find a good example for me I could understand and therefore don't know what to change. Sorry.

--- a/packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx
+++ b/packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx
@@ -1,0 +1,44 @@
+/*
+ * © 2021 Thoughtworks, Inc.
+ */
+
+import {
+  useGridSlotComponentProps,
+  GridToolbarContainer,
+} from '@material-ui/data-grid'
+import useStyles from './recommendationsTableStyles'
+import Pagination from '@material-ui/lab/Pagination'
+import { Box, MenuItem, Select, Typography } from '@material-ui/core'
+import { ReactElement } from 'react'
+
+function CustomPagination(
+  handlePageSizeChange: (event: React.ChangeEvent<{ value: unknown }>) => void,
+): ReactElement {
+  const { state, apiRef } = useGridSlotComponentProps()
+  const classes = useStyles()
+
+  return (
+    <GridToolbarContainer>
+      <Box display="flex" flexGrow={1} />
+      <Typography className={classes.rowsPerPage}>Rows per page:</Typography>
+      <Select value={state.pagination.pageSize} onChange={handlePageSizeChange}>
+        <MenuItem value={25}>25</MenuItem>
+        <MenuItem value={50}>50</MenuItem>
+        <MenuItem value={100}>100</MenuItem>
+      </Select>
+      <Pagination
+        className={classes.cell}
+        color="primary"
+        size="small"
+        shape="rounded"
+        count={state.pagination.pageCount}
+        page={state.pagination.page + 1}
+        showFirstButton
+        showLastButton
+        onChange={(event, value) => apiRef.current.setPage(value - 1)}
+      />
+    </GridToolbarContainer>
+  )
+}
+
+export default CustomPagination

--- a/packages/client/src/pages/RecommendationsPage/RecommendationsTable/RecommendationsTable.tsx
+++ b/packages/client/src/pages/RecommendationsPage/RecommendationsTable/RecommendationsTable.tsx
@@ -27,6 +27,7 @@ import Tooltip from 'common/Tooltip'
 import SearchBar from '../SearchBar'
 import Forecast from './Forecast/Forecast'
 import { Typography } from '@material-ui/core'
+import CustomPagination from './CustomPaginaton'
 
 type RecommendationsTableProps = {
   emissionsData: ServiceData[]
@@ -95,6 +96,12 @@ const RecommendationsTable: FunctionComponent<RecommendationsTableProps> = ({
   const [pageState, setPageState] = useState(initialPageState)
 
   const classes = useStyles()
+
+  const handlePageSizeChange = (
+    event: React.ChangeEvent<{ value: unknown }>,
+  ) => {
+    setPageState({ ...pageState, pageSize: event.target.value as number })
+  }
 
   const createRecommendationRows = (
     recommendations: RecommendationResult[],
@@ -209,9 +216,11 @@ const RecommendationsTable: FunctionComponent<RecommendationsTableProps> = ({
               }}
               onRowClick={handleRowClick}
               disableColumnFilter
-              pagination
               pageSize={pageState.pageSize}
-              rowsPerPageOptions={[25, 50, 100]}
+              components={{
+                Toolbar: () => CustomPagination(handlePageSizeChange),
+                Pagination: () => CustomPagination(handlePageSizeChange),
+              }}
               onPageSizeChange={(newPageSize) =>
                 setPageState({ ...pageState, pageSize: newPageSize })
               }

--- a/packages/client/src/pages/RecommendationsPage/RecommendationsTable/recommendationsTableStyles.ts
+++ b/packages/client/src/pages/RecommendationsPage/RecommendationsTable/recommendationsTableStyles.ts
@@ -41,6 +41,9 @@ const useStyles = makeStyles(() => ({
   recommendationsContainer: {
     marginTop: 65,
   },
+  rowsPerPage: {
+    paddingRight: '.5em',
+  },
 }))
 
 export default useStyles

